mesonpy/__init__.py

```diff
@@ -882,7 +884,6 @@ def build_commands(self, install_dir: Optional[pathlib.Path] = None) -> Sequence
             (
                 'meson',
                 'install',
-                '--only-changed',
```

Comment:
Can we keep this? It's quite important, at least verbose mode is completely unusable without it.

Reply:
It doesn't do anything anymore as of this PR. 

This is because we nuke the install dir before hand, so the destination file will never exist.

For reference, see https://github.com/mesonbuild/meson-python/pull/87#discussion_r1056438367.

Reply:
It's unfortunate that we have to do this, but the only other way, I think, is to patch meson, and I don't think we should leave editable installs broken in the meantime.

Reply:
To me, making the verbose mode effectively unusable seems worse than not cleaning up some now-unused old file (the chance of actual breakage is quite small here). So I'm ambivalent about this change.

That said, I don't use editable installs and there's further improvements in the pipeline, so I'll just register as neutral on this one; fine if others want to merge this.

Reply:
To clarify, this old file is still importable if the directory had an __init__.py, which can be confusing at the very least, and maybe cause problems(I'm thinking maybe if someone did a star import?).